unison-cli/src/Unison/Codebase/Editor/HandleInput/Update.hs

```diff
@@ -322,7 +323,9 @@ getSlurpResultForUpdate requestedNames = do
           --   [ (<#pingpong.pong + 1>, <Nat>),
           --     (<#pingpong.ping + 2>, <Nat>)
           --   ]
-          terms <- Codebase.unsafeGetTermComponent codebase oldHash
+          terms <-
+            Codebase.withConnection codebase \conn ->
+              Sqlite.runTransaction conn (Codebase.unsafeGetTermComponent codebase oldHash)
```

Comment:
I wonder if we could get much use out of a combinator that takes a `Codebase` and a `Transaction` and calls `Codebase.withConnection` and `Sqlite.runTransaction`.